pkg/dogstatsd/enrich.go

```diff
@@ -39,25 +40,31 @@ func extractTagsMetadata(tags []string, defaultHostname string, originTags strin
 	}
 	tags = tags[:n]
 
-	origin := ""
+	udsOrigin := ""
 	// We use the UDS socket origin if no origin ID was specify in the tags
 	// or 'dogstatsd_entity_id_precedence' is set to False (default false).
-	if entityIDValue == "" || !entityIDPrecedenceEnabled {
+	if originFromTag == "" || !entityIDPrecedenceEnabled {
 		// Add origin tags only if the entity id tags is not provided
-		origin = originTags
+		udsOrigin = originFromUDS
 	}
 
-	k8sOrigin := ""
-	// We set k8sOriginID if the metrics contain a 'dd.internal.entity_id' tag different from 'none'.
-	if entityIDValue != "" && entityIDValue != entityIDIgnoreValue {
+	// originFromClient can either be originFromTag or originFromMsg
+	originFromClient := ""
+
+	// We set originFromClient if the metrics contain a 'dd.internal.entity_id' tag different from 'none'.
+	if originFromTag != "" && originFromTag != entityIDIgnoreValue {
 		// Check if the value is not "none" in order to avoid calling
 		// the tagger for entity that doesn't exist.
 
 		// currently only supported for pods
-		k8sOrigin = kubelet.KubePodTaggerEntityPrefix + entityIDValue
+		originFromClient = kubelet.KubePodTaggerEntityPrefix + originFromTag
+	} else if originFromTag == "" && len(originFromMsg) > 0 {
+		// originFromMsg is the container id sent by the newer clients.
+		// Opt-in is handled when parsing.
+		originFromClient = containers.BuildTaggerEntityName(string(originFromMsg))
 	}
 
```

Comment:
global comment if you can put a matrix in the comment that explain who win if sever origin is provided it can help futur developer to understand 

```golang
| originFromTag | originFromUDS | originFromMsg | Result        |
|---------------|---------------|---------------|---------------|
| 0             | 0             | 0             | 0             |
| 0             | 0             | 1             |               |
| 0             | 1             | 0             | originFromUDS |
....


```